extensions/shared/register-jetpack-block.js

```diff
@@ -96,14 +96,18 @@ export default function registerJetpackBlock( name, settings, childBlocks = [] )
 		...settings,
 		title: buildBlockTitle( settings.title, buildBlockTags( name, requiredPlan ) ),
 		edit: requiredPlan ? wrapPaidBlock( { requiredPlan } )( settings.edit ) : settings.edit,
-		example: requiredPlan ? undefined : settings.example,
+		getEditWrapperProps() {
+			if ( requiredPlan ) {
+				return { 'data-upgrade-title': __( 'Upgrade required', 'jetpack' ) };
+			}
+		},
 	} );
 
 	if ( requiredPlan ) {
 		addFilter(
 			'editor.BlockListBlock',
-			`jetpack/${ name }-with-has-warning-is-interactive-class-names`,
-			withHasWarningIsInteractiveClassNames( `jetpack/${ name }` )
+			`jetpack/${ name }-with-custom-class-names`,
+			withCustomClassNames( `jetpack/${ name }`, 'has-warning is-interactive is-premium' )
```

Comment:
Maybe use `is-paid` instead of `is-premium`? I am not sure whether there are blocks requiring other non-Premium plans but can we be future proof here ❓

Reply:
Given that we're using `paid` in quite a few places, that's probably a good idea if only for consistency's sake. 👍

Reply:
keep in mind that it will cover only the blocks registered by Jetpack. Other blocks that require a paid plan are not bein handled by this. I've cherry-picked the previous commit of this PR and use this HOC but straight in the `./paid-blocks` folder. We are going to overlap effort :-/

Reply:
[Here](https://github.com/Automattic/jetpack/pull/16495/files#diff-efaec360257466dc6a630bca613fb7f1R26) is where I've added the custom classes, just FYI.